src/modules/adaptors/Convex/ConvexCurveAdaptor.sol

```diff
@@ -0,0 +1,333 @@
+    }
+
+    //============================================ Global Functions ===========================================
+    /**
+     * @dev Identifier unique to this adaptor for a shared registry.
+     * Normally the identifier would just be the address of this contract, but this
+     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
+     * of the adaptor is more difficult.
+     */
+    function identifier() public pure virtual override returns (bytes32) {
+        return keccak256(abi.encode("Convex Curve Adaptor V 0.0"));
+    }
+
+    //============================================ Implement Base Functions ===========================================
+
+    /**
+     * @notice Deposit & Stakes LPT from the cellar into Convex Market at the end of the user deposit sequence.
+     * @param assets amount of LPT to deposit and stake
+     * @param adaptorData see adaptorData info at top of this smart contract
+     */
+    function deposit(uint256 assets, bytes memory adaptorData, bytes memory) public override {
+        (uint256 pid, address rewardsPool, ERC20 lpt, CurvePool pool, bytes4 selector) = abi.decode(
+            adaptorData,
+            (uint256, address, ERC20, CurvePool, bytes4)
+        );
+
+        if (selector != bytes4(0)) _callReentrancyFunction(pool, selector);
+        else revert BaseAdaptor__UserDepositsNotAllowed();
+
+        _validatePositionIsUsed(pid, rewardsPool, lpt, pool, selector);
+        lpt.safeApprove(address(booster), assets);
+
+        booster.deposit(pid, assets, true);
+
+        // Zero out approvals if necessary.
+        _revokeExternalApproval(lpt, address(booster));
+    }
+
+    /**
+     * @notice If a user withdraw needs more LPTs than what is in the Cellar's wallet, then the Cellar will unstake, unwrap cvxLPTs, and withdraw LPTs from Convex
+     * @param amount of LPT to unstake, unwrap, and withdraw from Convex market
+     * @param receiver see baseAdaptor.sol
+     * @param adaptorData see adaptorData info at top of this smart contract
+     * @param configurationData see configurationData at top of this smart contract
+     */
+    function withdraw(
+        uint256 amount,
+        address receiver,
+        bytes memory adaptorData,
+        bytes memory configurationData
+    ) public override {
+        // Check that position is setup to be liquid.
+        bool isLiquid = abi.decode(configurationData, (bool));
+
+        // Run external receiver check.
+        _externalReceiverCheck(receiver);
+
+        (uint256 pid, address rewardPool, ERC20 lpt, CurvePool pool, bytes4 selector) = abi.decode(
+            adaptorData,
+            (uint256, address, ERC20, CurvePool, bytes4)
+        );
+
+        if (isLiquid && selector != bytes4(0)) _callReentrancyFunction(pool, selector);
+        else revert BaseAdaptor__UserWithdrawsNotAllowed();
+
+        _validatePositionIsUsed(pid, rewardPool, lpt, pool, selector);
```

Comment:
Can you move this line above the call reentrancy line?

Reply:
Done